cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -459,17 +459,11 @@ func (s *Supervisor) startOpAMPClient() error {
 func (s *Supervisor) startOpAMPServer() error {
 	s.opampServer = server.New(newLoggerFromZap(s.logger))
 
-	var err error
-	s.opampServerPort, err = s.findRandomPort()
```

Comment:
Can you explain this change?

Reply:
When I was debugging issues on windows, it seemed like the agent wasn't able to connect because the server port in the agent config wasn't matching the port the supervisor was listening on. That's because the config is already constructed before the opamp server starts.

Basically the flow was:
1. getBootstrapInfo determines opamp port -> generate initial config -> startOpAMPServer determines opamp port.

It feels like we should only determine the opamp server port once initially, which is what this change is.

Reply:
Nice catch. You're right, we're setting it twice. I'm good with this, thanks for the explanation.